packages/eventual-send/src/handled-promise.js

```diff
@@ -439,6 +443,7 @@ export const makeHandledPromise = () => {
       }
       // Assimilate the `resolvedPromise` as an actual frozen Promise, by
       // treating `resolvedPromise` as if it is a non-promise thenable.
+      // this is the same type as the Promise constructor
```

Comment:
I'm pretty sure you don't mean "the Promise constructor", which is `new Promise(executor)` and only takes a single argument.  Would the following typing be better?
```suggestion
      // @type {Promise['then']}
```

Reply:
oops that was me attempting to get an LLM to annotate the type. And yeah that was not accurate anyway.

I think your suggestion will work. I'll try it and if not I'll just remove any attempt